Tools/ci-run.sh

```diff
@@ -3,125 +3,176 @@
     export CXX="g++"
   fi
-fi
-if [ "${OS_NAME##macos*}" == "" ]; then
+elif [[ $OSTYPE == "darwin"* ]]; then
+  echo "Setting up macos compiler"
   export CC="clang -Wno-deprecated-declarations"
   export CXX="clang++ -stdlib=libc++ -Wno-deprecated-declarations"
 fi
 
 # Set up miniconda
-if [ "$STACKLESS" == "true" ]; then
+if [[ $STACKLESS == "true" ]]; then
   echo "Installing stackless python"
   #conda install --quiet --yes nomkl --file=test-requirements.txt --file=test-requirements-cpython.txt
   conda config --add channels stackless
   conda install --quiet --yes stackless || exit 1
 fi
 
+PYTHON_SYS_VERSION=$(python -c 'import sys; print(sys.version)')
+
 # Log versions in use
 echo "===================="
 echo "|VERSIONS INSTALLED|"
 echo "===================="
-python -c 'import sys; print("Python %s" % (sys.version,))'
-if [ "$CC" ]; then
+echo "Python $PYTHON_SYS_VERSION"
+if [[ $CC ]]; then
   which ${CC%% *}
   ${CC%% *} --version
 fi
-if [ "$CXX" ]; then
+if [[ $CXX ]]; then
   which ${CXX%% *}
   ${CXX%% *} --version
 fi
 echo "===================="
 
 # Install python requirements
 echo "Installing requirements [python]"
-if [ -z "${PYTHON_VERSION##2.7}" ]; then
+if [[ $PYTHON_VERSION == "2.7"* ]]; then
   pip install wheel || exit 1
   pip install -r test-requirements-27.txt || exit 1
-elif [ -z "${PYTHON_VERSION##3.[45]*}" ]; then
+elif [[ $PYTHON_VERSION == "3."[45]* ]]; then
   python -m pip install wheel || exit 1
   python -m pip install -r test-requirements-34.txt || exit 1
 else
   python -m pip install -U pip setuptools wheel || exit 1
 
-  if [ -n "${PYTHON_VERSION##*-dev}" ]; then
+  if [[ $PYTHON_VERSION != *"-dev" || $COVERAGE == "1" ]]; then
     python -m pip install -r test-requirements.txt || exit 1
 
-    if [ "${PYTHON_VERSION##pypy*}" -a "${PYTHON_VERSION##3.[4789]*}" ]; then
+    if [[ $PYTHON_VERSION != "pypy"* && $PYTHON_VERSION != "3."[4789]* ]]; then
       python -m pip install -r test-requirements-cpython.txt || exit 1
     fi
   fi
 fi
 
-if [ "$TEST_CODE_STYLE" == "1" ]; then
-  STYLE_ARGS="--no-unit --no-doctest --no-file --no-pyregr --no-examples";
+if [[ $TEST_CODE_STYLE == "1" ]]; then
+  STYLE_ARGS="--no-unit --no-doctest --no-file --no-pyregr --no-examples"
   python -m pip install -r doc-requirements.txt || exit 1
 else
-  STYLE_ARGS="--no-code-style";
+  STYLE_ARGS="--no-code-style"
 
   # Install more requirements
-  if [ -n "${PYTHON_VERSION##*-dev}" ]; then
-    if [ -z "${BACKEND##*cpp*}" ]; then
+  if [[ $PYTHON_VERSION != *"-dev" ]]; then
+    if [[ $BACKEND == *"cpp"* ]]; then
       echo "WARNING: Currently not installing pythran due to compatibility issues"
       # python -m pip install pythran==0.9.5 || exit 1
-    fi
-
-    if [ "$BACKEND" != "cpp" -a -n "${PYTHON_VERSION##pypy*}" -a -n "${PYTHON_VERSION##2*}" -a -n "${PYTHON_VERSION##*3.4}" ]; then
+    elif [[ $PYTHON_VERSION != "pypy"* && $PYTHON_VERSION != "2"* && $PYTHON_VERSION != *"3.4" ]]; then
```

Comment:
Originally (`"$BACKEND" != "cpp"`) the last conditional will run if `BACKEND == "c" / "c,cpp"`, now it will run only if `BACKEND == "c"`. @scoder was it an error in the original and therefore should we change this behavior back?

Reply:
`!= "cpp"` was correct. It's only intended to avoid useless duplication when splitting jobs into C/C++.

Reply:
OTOH, the star in `*"3.4"` seems useless.

Reply:
> OTOH, the star in `*"3.4"` seems useless.

It is.
It were in the original i i suppose that it should be `"3.4"*`